flare/framework/computation_data_processor.py

```diff
@@ -113,8 +115,11 @@ def _training_loop(self):
         agent_ids = []
         data = []
         while not self.exit_flag:
+            num_agents = sum([r.value for r in self.agent_runnings])
+            if num_agents == 0:
+                return
             try:
-                while len(agent_ids) < self.num_agents:
+                while len(agent_ids) < num_agents:
```

Comment:
what if between line 118 and 122, another `agent` stops? The `while` loop may never exit.